src/lib/tabs/tabWarp/WarpCalculatorTab.tsx

```diff
@@ -49,6 +50,46 @@ function Inputs() {
 
   const warpRequest = sanitizeWarpRequest(storedWarpRequest)
 
+  scannerChannel.use((event) => {
+    console.log(event)
+    switch (event.event) {
+      case "UpdateGachaFunds":
+        form.setFieldValue("jades", event.data.stellar_jade + event.data.oneric_shards)
+        break
+
+      case "UpdateMaterials":
+        const state = useScannerState.getState()
+        const specialPasses = state.materials["102"] ?? { count: 0 }
+        const undyingStarlight = state.materials["252"] ?? { count: 0 }
+
+        form.setFieldValue("passes", specialPasses.count + Math.floor(undyingStarlight.count / 20))
+        break
+      
+      case "GachaResult":
+        const gachaResult = event.data
+        const pityUpdate = gachaResult.pity_5
+
+        if (gachaResult.banner_type == "Character") {
+          if (pityUpdate.kind == "ResetPity") {
+            form.setFieldValue("pityCharacter", pityUpdate.amount)
+            form.setFieldValue("guaranteedCharacter", pityUpdate.set_guarantee)
+          } else { // AddPity
+            const currentPity = form.getFieldValue("pityCharacter")
+            form.setFieldValue("pityCharacter", currentPity + gachaResult.pity_5.amount)
+          }
+
+        } else if (gachaResult.banner_type == "LightCone") {
+          if (pityUpdate.kind == "ResetPity") {
+            form.setFieldValue("pityLightCone", pityUpdate.amount)
+            form.setFieldValue("guaranteedLightCone", pityUpdate.set_guarantee)
+          } else { // AddPity
+            const currentPity = form.getFieldValue("pityLightCone")
+            form.setFieldValue("pityLightCone", currentPity + gachaResult.pity_5.amount)
+          }
+        } // TODO: Handle Standard Banner?
```

Comment:
> // TODO: Handle Standard Banner?

I would defer for waaay low priority, I've never seen anyone care about standard pity lmao

Reply:
Yeah, hence the question mark haha. We can just remove the TODO altogether though.